include/xeus-lite/xembind.hpp

```diff
@@ -40,17 +40,26 @@ namespace xeus
 
     xeus::xserver * get_server(xeus::xkernel * kernel);
 
-    template<class interpreter_type>
-    std::unique_ptr<xkernel> make_xkernel()
+    template <class interpreter_type>
+    using builder_type = interpreter_type* (*)(ems::val);
+
+    template <class interpreter_type>
+    interpreter_type* default_builder(ems::val)
+    {
+        return new interpreter_type();
+    }
+
+    template<class interpreter_type, builder_type<interpreter_type> builder>
+    std::unique_ptr<xkernel> make_xkernel(ems::val js_argv)
```

Comment:
I suppose we don't want `js_argv` as parameter here but that will be up to the xeus-cpp interpreter builder to pass argv?

Reply:
Hmmm, So make_xkernel is our factory function here. We're using it something like the following (read the second way of how a constructor can be framed here
https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html#embind-external-constructors)

So raw_xkernel = new Module.xkernel(argv); -> directly maps to make_xkernel(ems::val js_argv) 

And so we don't mention the parameters for make_xkernel in the constructor

Reply:
We actually even tried just passing the built interpreter to make_xkernel but I couldn't put that to work because of this https://github.com/jupyter-xeus/xeus-lite/pull/16#issuecomment-2837266397

Reply:
Not really, these arguments are passed from the javascript to the kernel constructor which in fine calls this method.

Reply:
(Sorry I missed the previous replies when answering, the page was not refreshed)

Reply:
Yeah so it changes the JS API.

My point of the comment is that passing argv here is specific to xeus-cpp. Other kernels don't need that (yet?). 

I thought the point of making configurable the interpreter_builder was to have in xeus-cpp a specific builder that takes argv into account?

Reply:
Yeah so for other kernels where the args can be ignored, this is the default builder that would play a role 

https://github.com/jupyter-xeus/xeus-lite/pull/16/files#diff-b59174e2f9d6c1c5c9d8b494814d5855997f1926c03e4462914e1039f9cb60ebR47

So yeah we just ignore them for the kernels that won't need them !

Reply:
Ok, so no change required on any other kernel or jupyterlite-xeus then? Ship it!

Reply:
Just a one liner 

https://github.com/jupyterlite/xeus/pull/210